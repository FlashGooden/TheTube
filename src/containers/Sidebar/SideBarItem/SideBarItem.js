import React from 'react'
import {Icon,Menu} from 'semantic-ui-react'
import './SideBarItem.scss'

export default function SideBarItem(props) {
  const highlight = props.highlight ? 'highlight-item' : null;
  return (
    <Menu.item className={['sidebar-item', highlight].join('')}>
      <div className='sidebar-item-alignment-container'>

      </div>
    </Menu.item>
  )
}
