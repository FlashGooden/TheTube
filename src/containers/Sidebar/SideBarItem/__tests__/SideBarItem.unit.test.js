import React from 'react'
import {shallow} from 'enzyme'
import {SideBarItem} from '../SideBarItem'
import {Menu,} from 'semantic-ui-react'

describe('SideBarItem', ()=> {
  test('should render Sidebar', ()=> {
    const wrapper = shallow(
         <SideBarItem/>   
    )
    expect(wrapper).toMatchSnapshot()
  })
  
  test('renders highlighted sideitem', ()=> {
    const wrapper = shallow(
      <SideBarItem highlight={true} icon='fire' label='trending' />
    )
    expect(wrapper).toMatchSnapshot()
  }) 
  
  test('render Non Highlighted bar', ()=> {
    const wrapper = shallow(
      <SideBarItem icon='fire' label='trending'/>
    )
    expect(wrapper).toMatchSnapshot()
  
    // expect(warpper)
  })
})
