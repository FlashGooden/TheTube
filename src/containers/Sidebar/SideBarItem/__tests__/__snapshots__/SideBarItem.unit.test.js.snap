// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem render Non Highlighted bar 1`] = `
<MenuItem
  className="sidebar-item"
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem renders highlighted sideitem 1`] = `
<MenuItem
  className="sidebar-itemhighlight-item"
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem should render Sidebar 1`] = `
<MenuItem
  className="sidebar-item"
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;
